compiler/rustc_hir_analysis/src/structured_errors/wrong_number_of_generic_args.rs

```diff
@@ -728,7 +728,10 @@ impl<'a, 'tcx> WrongNumberOfGenericArgs<'a, 'tcx> {
         && let Some(trait_path_segment) = path.segments.get(0) {
             let num_generic_args_supplied_to_trait = trait_path_segment.args().num_generic_params();
 
-            if num_assoc_fn_excess_args == num_trait_generics_except_self - num_generic_args_supplied_to_trait {
+            if num_trait_generics_except_self
+                .checked_sub(num_generic_args_supplied_to_trait)
+                .is_some_and(|diff| diff == num_assoc_fn_excess_args)
```

Comment:
Is it possible to just do this transformation: `a = b - c` <=> `a + c = b`?

```suggestion
            if num_generic_args_supplied_to_trait + num_assoc_fn_excess_args == num_trait_generics_except_self
```